src/usePopper.js

```diff
@@ -98,35 +98,32 @@ export const usePopper = (
   ]);
 
   const popperInstanceRef = React.useRef();
-  const createPopper = React.useMemo(
-    () => options.createPopper || defaultCreatePopper,
-    [options.createPopper]
-  );
 
   useIsomorphicLayoutEffect(() => {
-    let popperInstance = null;
-    if (referenceElement != null && popperElement != null) {
-      popperInstance = createPopper(
-        referenceElement,
-        popperElement,
-        popperOptions
-      );
-
-      popperInstanceRef.current = popperInstance;
+    if (popperInstanceRef.current) {
+      popperInstanceRef.current.setOptions(popperOptions);
+    }
+  }, [popperOptions]);
+
+  useIsomorphicLayoutEffect(() => {
+    if (referenceElement == null || popperElement == null) {
+      return;
```

Comment:
We can return early here, that way we don't schedule a cleanup function that is useless when we don't create a popper instance.